examples/zonal_private_iap_bastion/outputs.tf

```diff
@@ -0,0 +1,132 @@
+}
+
+output "endpoint" {
+  sensitive   = true
+  description = "Cluster endpoint"
+  value       = "${module.gke.endpoint}"
+}
+
+output "min_master_version" {
+  description = "Minimum master kubernetes version"
+  value       = "${module.gke.min_master_version}"
+}
+
+output "logging_service" {
+  description = "Logging service used"
+  value       = "${module.gke.logging_service}"
+}
+
+output "monitoring_service" {
+  description = "Monitoring service used"
+  value       = "${module.gke.monitoring_service}"
+}
+
+output "master_authorized_networks_config" {
+  description = "Networks from which access to master is permitted"
+  value       = "${module.gke.master_authorized_networks_config}"
+}
+
+output "master_version" {
+  description = "Current master kubernetes version"
+  value       = "${module.gke.master_version}"
+}
+
+output "ca_certificate" {
+  sensitive   = true
+  description = "Cluster ca certificate (base64 encoded)"
+  value       = "${module.gke.ca_certificate}"
+}
+
+output "network_policy_enabled" {
+  description = "Whether network policy enabled"
+  value       = "${module.gke.network_policy_enabled}"
+}
+
+output "http_load_balancing_enabled" {
+  description = "Whether http load balancing enabled"
+  value       = "${module.gke.http_load_balancing_enabled}"
+}
+
+output "horizontal_pod_autoscaling_enabled" {
+  description = "Whether horizontal pod autoscaling enabled"
+  value       = "${module.gke.horizontal_pod_autoscaling_enabled}"
+}
+
+output "kubernetes_dashboard_enabled" {
+  description = "Whether kubernetes dashboard enabled"
+  value       = "${module.gke.kubernetes_dashboard_enabled}"
+}
+
+output "node_pools_names" {
+  description = "List of node pools names"
+  value       = "${module.gke.node_pools_names}"
+}
+
+output "node_pools_versions" {
+  description = "List of node pools versions"
+  value       = "${module.gke.node_pools_versions}"
+}
+
+output "service_account" {
+  description = "The service account to default running nodes as if not overridden in `node_pools`."
+  value       = "${module.gke.service_account}"
+}
+
+output "network_name" {
+  value       = "${module.gke-network.network_name}"
+  description = "The name of the VPC being created"
+}
+
+output "get_credentials" {
+  description = "Gcloud get-credentials command"
+  value       = format("gcloud container clusters get-credentials --project %s --zone %s --internal-ip %s", var.project_id, module.gke.location, var.cluster_name)
+}
+
+output "bastion_ssh" {
+  description = "Gcloud compute ssh to the bastion host command"
+  value       = format("gcloud beta compute ssh %s --tunnel-through-iap --project %s --zone %s -- -L8888:127.0.0.1:8888", module.bastion.hostname, var.project_id, local.bastion_zone)
+}
+
+output "bastion_kubectl" {
+  description = "kubectl command using the local proxy once the bastion_ssh command is running"
+  value       = "HTTPS_PROXY=localhost:8888 kubectl get pods --all-namespaces"
```

Comment:
https://github.com/GoogleContainerTools/skaffold/issues/3254

That is pretty much the problem.  I want to use skaffold to deploy stuff, but it runs both gcloud and kubectl commands, and one needs http proxy and the other must not use it.

Reply:
In truth, the generic http proxy method for accessing a private cluster through a bastion doesn't seem to really work all that well.  At least on OS X, the system-wide proxy settings also have only a no-proxy setting, rather than an only-proxy setting, so it is going to be an exercise in constantly fighting against that proxy configuration just for the endpoint addreess until I do something drastic like run a configurable proxy locally, which only forwards certain addresses to the ssh tunnel, or I figure out how to get an actual VPN connection to my VPC without having to buy a VPN-capable router.  There has to be some kind of software (preferably open-source software) solution to the problem of connecting to a VPC network from a single host in a coffee shop or airport.

Th other alternative is to set up an ssh tunnel directly to the private endpoint, and then configure kubectl to access the server at 127.0.0.1, and let the ssh tunnel translate that to the correct host.  But that can break things, too, if kubectl ever returns values that include the endpoint address without using the value received in the Host header.